### PR TITLE
pkg/server: encode json straight to response

### PR DESCRIPTION
**Pull request by runcom**
Signed-off-by: Antonio Murdaca <runcom@linux.com>

**- What I did**

Avoid marshaling into memory as well and remove not needed
content-length sets.

Should help with #519 

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Comment by cgwalters** [comment]
Is this for 4.0?  Can you add the label if so?

LGTM though I find myself wondering if the CPU usage is related to the ELB health checks (`/healthz`).  Looks like the interval there is every 10 seconds.

Machines only pull from this endpoint when they're booting.

Going to
/assign LorbusChris
for another review.

File: pkg/server/api.go [hunk 1]
```diff
@@ -79,13 +79,11 @@ func NewServerAPIHandler(s Server) *APIHandler {
 // API handler.
 func (sh *APIHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	if r.Method != http.MethodGet && r.Method != http.MethodHead {
-		w.Header().Set("Content-Length", "0")
```
Comment (LorbusChris) [inline pkg/server/api.go]: Tests expect content length to be 0 in error cases. If we don't encode content length at all, tests need to be adapted.

File: pkg/server/api.go [hunk 2]
```diff
@@ -107,24 +104,15 @@ func (sh *APIHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
 
-	data, err := json.Marshal(conf)
-	if err != nil {
-		w.Header().Set("Content-Length", "0")
-		w.WriteHeader(http.StatusInternalServerError)
-		glog.Errorf("failed to marshal %v config: %v", cr, err)
-		return
-	}
-
-	w.Header().Set("Content-Length", fmt.Sprintf("%d", len(data)))
```
Comment (LorbusChris) [inline pkg/server/api.go]: why was this done before?

Comment (runcom) [reply pkg/server/api.go]: I don't know, but if you write something to the response writer which has a length, that header is automatically added with the correct length

File: pkg/server/api.go (hunk 1 above)
Comment (runcom) [reply pkg/server/api.go]: Yup, on it

**Comment by runcom** [comment]
closing this for now as it's not properly working as I've expected

**Comment by runcom** [comment]
> though I find myself wondering if the CPU usage is related to the ELB health checks (`/healthz`).

That's served by the MCD though and MCD isn't showing high cpu usage

**Pull request closed without merging**